### PR TITLE
Add GetInfo() support for solaris

### LINKED ISSUE #5173: OmniOS: minio server fails to start "Initializing object layer failed cause=disk path full"
## Expected Behavior
Minio server should start and use the ZFS filesystem for it's storage location.

## Current Behavior
On OmniOS (SunOS) version omnios-r151022, with a newly created ZFS filesystem, minio server fails to start with:
`minio server /minio
ERRO[0000] Initializing object layer failed              cause=disk path full source=[server-main.go:209:serverMain()]

There's plenty of space available:
Filesystem             Size   Used  Available Capacity  Mounted on
rpool/minio            899G    88K       446G     1%    /minio

Trussing the process, the last call I see before the error is thrown:
  stat("/minio", 0xC42042FB38)                    = 0

## Possible Solution
Searching on the error message, I found https://github.com/minio/minio/issues/3592. I suspect that the issue here is the same, because ZFS does not preallocate inodes -- so long as there is space, metadata can be written. For example, df -oi (equivalent of df -i on Linux) returns "df: operation not applicable for FSType zfs".

## Steps to Reproduce (for bugs)
1. Compile from source on OmniOSCE omnios-r151022-eb9d5cb557, using go ver 1.9.2
2. Create new ZFS filesystem called minio
3. Attempt to start minio server, using default configuration and no command-line options: `minio server /minio`

## Context
I'm trying to run minio on my OmniOS file server to serve as a backup target for Duplicati.

## Your Environment
* `minio version` doesn't appear to be functioning correctly:
Version: DEVELOPMENT.GOGET
Release-Tag: DEVELOPMENT.GOGET
Commit-ID: DEVELOPMENT.GOGET
From minio.spec:
RELEASE.2017-04-25T01-27-49Z
83abb310b4ce3a0dfc6d7faf78e33cb6f9132cfe
* System OS: OmniOS omnios-r151022-eb9d5cb557

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add GetInfo() support for solaris

## Motivation and Context
Fixes #5173

## How Has This Been Tested?
Manually

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

File: pkg/disk/type_bsd.go [hunk 1]
```diff
@@ -19,9 +19,9 @@
 package disk
 
 // getFSType returns the filesystem type of the underlying mounted filesystem
-func getFSType(fstype [16]int8) string {
-	b := make([]byte, len(fstype[:]))
-	for i, v := range fstype[:] {
+func getFSType(fstype []int8) string {
```
Comment (balamurugana) [inline pkg/disk/type_bsd.go:22]: previously `fstype` is exact size.  is it changed now?

Comment (harshavardhana) [reply pkg/disk/type_bsd.go:22]: Yes we don't it to be. Solaris has [32]int8 so made this function more generic.

Comment (balamurugana) [reply pkg/disk/type_bsd.go:22]: OK. Now this function is doing generic `int8SliceToString()`.  This change can be a separate PR.

**Review by balamurugana — APPROVED** [review]
LGTM

**Review by krisis — APPROVED** [review]

**Pull request merged**